### PR TITLE
Lib2/fix/async pixel diffs

### PR DESCRIPTION
**Pull request by a-miscellaneous**
changed IO blocking to CPU blocking
asynced pixell diff calculation.
the profiler lies:
old Version is 53 seconds:
Screenshot_263
Screenshot_261

the new version is apperently 71 seconds, while the whole calculation is under 50s  ¯\\_(ツ)_/¯:

Screenshot_268
Screenshot_264

### CODE REVIEW

**Comment by akriese** [comment]
I think the weird times by the profiler might come from multithreading. It might not actually show the wallclock time. CPU time is all the threads' time combined.

**Comment by akriese** [comment]
@a-miscellaneous why is Dispatchers.Default prefered in 0b08ac1?

**Comment by akriese** [comment]
You can squash the last three commits into one :) Tell me if you need help with that

**Comment by a-miscellaneous** [comment]
> @a-miscellaneous why is Dispatchers.Default prefered in [0b08ac1](https://github.com/amosproj/amos2023ws03-gui-frame-diff/commit/0b08ac18e0d64da9edf5adb5afa54a1cc436d69c)?

there is .Main -> main thread aka UI
there is .IO -> threads for input and output, 
and there is .Default which is general CPU.
We should have probably never used IO in the first place

File: DifferenceGenerator/build.gradle.kts [hunk 1]
```diff
@@ -21,6 +21,8 @@ dependencies {
     testImplementation(kotlin("test"))
     implementation("org.bytedeco:javacv-platform:1.5.7")
     implementation(project(path = ":VideoGenerator"))
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.8.0-RC")
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-swing:1.7.1")
```
Comment (akriese) [inline DifferenceGenerator/build.gradle.kts:25]: Do we need both of these?

**Review by akriese — APPROVED** [review]

File: DifferenceGenerator/build.gradle.kts (hunk 1 above)
Comment (a-miscellaneous) [reply DifferenceGenerator/build.gradle.kts]: actually no

**Pull request merged**